### PR TITLE
Add documentation builder to devel docs

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

**Comment by pmatulis** [comment]
affects #2047

**Comment by anthonydillon** [comment]
@evilnick @pmatulis I have applied all the requested changes. Could you please take a look at this one.

Once the backports are complete I'll create a new PR to include versions.

**Comment by anthonydillon** [comment]
@evilnick @pmatulis I have tested this branch on jujucharms.com locally and it works. Therefore, proving that these changes are backwards compatible.

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,3185 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
[the remark this answers is not in the document]
Comment (evilnick) [reply yarn.lock]: ok

**Pull request merged**